libs/wingsdk/src/std/array.ts

```diff
@@ -0,0 +1,65 @@
+// These classes are used by Wing to provide JSII subsets of the JS Array class.
+// They should not be consumed directly by users.
+// TODO: These should be interfaces, currently Wing does not support interface JSII imports
+
+/**
+ * Mutable Array
+ */
```

Comment:
Thought...

Instead of using `any` as a placeholder by convention, can we use a placeholder type. Something like this:

```js
/**
 * This type is not going to be visible in Wing. It is just used
 * as a placeholder for type parameters. In TypeScript it is `== any`.
 * 
 * @placeholder
 */
export class T {}

/**
 * @typeparam T
 */
export class MutArray {
  push(item: T): void;
}
```

We export `T` as a "placeholder type" and then we add an annotation to `MutArray` that tells Wing that the class takes a type parameter. Then, when we import the type into Wing, we can replace `T` instead of `any`.

Does it make any sense?

Reply:
Yeah that makes perfect sense, and I considered this (mentioned in another comment) but I decided on the simpler approach for now because this doesn't buy us anything yet. This would however be required if we want to support multiple type args. I'm on the fence about whether or not to include it in this PR, what do you think?

Reply:
Discussed offline, decided to table this for now.